src/TasksModule/Components/TasksBoard/TasksBoard.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useContext, useEffect, useState } from "react";
+import styles from "./TasksBoard.module.css";
+import axios from "axios";
+import { baseUrl } from "../../../Constants/Components/Urls";
+import { motion } from "framer-motion";
+import { AuthContext } from "../../../Context/Components/AuthContext.js";
+
+
+export default function TasksBoard() {
+  const token = localStorage.getItem("userToken");
+  const {userRole}:string=useContext(AuthContext);
+  const [toDoTasks, setToDoTasks] = useState<object[]>();
```

Comment:
Having the type as `object[]` is not beneficial for us, we need to put the correct types, to be able to detect errors in our applications and get autocompletion.

Reply:
We don't need 3 state variables for each task, we can have just one that will contain all of them and then have a normal variable that we store the corresponding tasks:
```js
const [tasks, setTasks] = useState<Task[]>([]);

const todos : Task[]= tasks.filter(({status})=> status ==="ToDo")
```

and the same for other ones.

Reply:
You can read more about having a derived state here from the [React Docs](https://react.dev/learn/managing-state#choosing-the-state-structure)